osu.Server.Spectator/Hubs/TeamVsRuleset.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Linq;
+using osu.Game.Online.Multiplayer;
+using osu.Game.Online.Multiplayer.MatchRulesets.TeamVs;
+
+namespace osu.Server.Spectator.Hubs
+{
+    public class TeamVsRuleset : MatchRuleset
+    {
+        private readonly TeamVsMatchRoomState state;
+
+        public TeamVsRuleset(ServerMultiplayerRoom room)
+            : base(room)
+        {
+            room.MatchRulesetState = state = TeamVsMatchRoomState.CreateDefault();
+
+            room.UpdateMatchRulesetRoomState(room);
+        }
+
+        public override void HandleUserJoined(MultiplayerRoomUser user)
+        {
+            base.HandleUserJoined(user);
+
+            // if the state already has a team specification we can reuse it (and keep the user's chosen team).
+            if (user.MatchRulesetState is TeamVsMatchUserState)
+                return;
```

Comment:
Probably yeah. The idea was to show that this could potentially retain team choice between ruleset selection, but I'll just remove for now as it's not required for any existing functionality.